d3a_api_client/rest_device.py

```diff
@@ -83,34 +84,42 @@ def set_energy_forecast(self, pv_energy_forecast_Wh, do_not_wait=False):
             return self.dispatcher.wait_for_command_response('set_energy_forecast', transaction_id)
 
     @logging_decorator('offer')
-    def offer_energy(self, energy, price):
-        transaction_id, posted = self._post_request('offer', {"energy": energy, "price": price})
+    def offer_energy(self, energy: float, price: float, replace_existing: bool = True):
+        transaction_id, posted = self._post_request(
+            'offer', {"energy": energy, "price": price, 'replace_existing': replace_existing})
+
         if posted:
             response = self.dispatcher.wait_for_command_response('offer', transaction_id)
             log_bid_offer_confirmation(response)
             return response
 
     @logging_decorator('offer')
-    def offer_energy_rate(self, energy, rate):
+    def offer_energy_rate(self, energy: float, rate: float, replace_existing: bool = True):
         transaction_id, posted = self._post_request(
-            'offer', {"energy": energy, "price": rate * energy})
+            'offer', {
+                "energy": energy, "price": rate * energy, 'replace_existing': replace_existing})
```

Comment:
could you please use this opportunity to homogenise the string quotes to `'`? Thanks!

Reply:
Done!